godtossing/Pos/Ssc/GodTossing/Arbitrary.hs

```diff
@@ -103,18 +107,22 @@ commitmentsAndOpenings =
     vssPk = toVssPublicKey $ deterministicVssKeyGen "ababahalamaha"
 {-# NOINLINE commitmentsAndOpenings #-}
 
-
 instance Arbitrary CommitmentOpening where
     arbitrary = elements commitmentsAndOpenings
+    shrink = genericShrink
 
 instance Nonrepeating CommitmentOpening where
     nonrepeating n = sublistN n commitmentsAndOpenings
 
 instance Arbitrary Commitment where
     arbitrary = coCommitment <$> arbitrary
+    shrink Commitment {..} = [ Commitment { commShares = shrunkShares, .. }
```

Comment:
I still expect a comment for each non-generic implementation. Here I want to see why we're not shrinking `vcSigningKey`, `vcVssKey` and `vcExpiryEpoch` (because of the invariant).